### ISSUE TITLE
Run SEN12MS baseline

### ISSUE THREAD

**Issue by taeil** [Berkeley-Data/irrigation_detection#97]
https://github.com/schmitt-muc/SEN12MS

**Comment by suryatechie** [comment]
Looking into this issue.

**Comment by suryatechie** [comment]
Code is using tensorboard. It takes logs directory as input to write the tensorboard logs to be retrieved by the tensorboard server to display the metrics. Launched the tensorboard container using the following steps:
1) Created logs folder under root (/logs)
2) Run the tensorbpard docker container to avoid the installation of tensorboard packages on the main OS.
docker run --name tnsb -d --rm -p 6006:6006 -v /your/log/dir:/root/logs tensorflow/tensorflow:latest tensorboard --bind_all --logdir /root/logs
3) Access from the browser **http://44.231.248.158:6006/**

**Comment by suryatechie** [comment]
**Train the model**

1. cd ~/SEN12MS
2. docker run --name tf2 --gpus all -it --rm -p 8888:8888 -v $PWD:/workspace/app -v /tmp:/tmp -v $PWD/../data/processed:/workspace/app/data tf2
3. cd classification
4. pip install -r requirements.txt
5. Copy IGBP_probability_labels.pkl from labels to splits folder as main_train.py is looking for it under splits folder  
cp /workspace/app/labels/IGBP_probability_labels.pkl /workspace/app/splits/IGBP_probability_labels.pkl  
7. CUDA_VISIBLE_DEVICES=0 python main_train.py --exp_name sem12ms_baseline --data_dir /workspace/app/data/sen12ms --label_split_dir /workspace/app/splits --use_RGB --IGBP_simple --label_type multi_label --threshold 0.1 --model DenseNet121 --lr 0.001 --decay 1e-5 --batch_size 64 --num_workers 4 --epochs 1

**Comment by suryatechie** [comment]
Found the rootcause. The train_list_updated.txt has '\n' characters at the end of each line. Trimmed those using rstrip('\n') and regenerated the pickle files. It started working!

**Comment by suryatechie** [comment]
**For 1 epochs:**
Validation microPrec: 0.693667 microF1: 0.539214 sampleF1: 0.495452 microF2: 0.475667 sampleF2: 0.479950
I Will work on improving. Also, need to fix the tensorboard log directory path.

**Comment by suryatechie** [comment]
Baseline to be reproduced
![image](https://user-images.githubusercontent.com/15736461/109337210-51fd4380-7819-11eb-868b-789f1baa3944.png)

**Comment by taeil** [comment]
I will change to wandb instead of tensorboard today or tomorrow.

**Comment by suryatechie** [comment]
**Baseline evaluation based on the Paper**
**First network (image classification using ResNet-110)** 
**Input:** Summer subset with cut images 64x64 from S2 samples (10 bands)
**Test data:** City of Munich

**Second network (image segmentation using DenseNet)**
**Input:** S2 samples with full size images 256x256 (10 bands)
**Test data:** City of Rome

**Comment by taeil** [comment]
[running 50 epochs ](https://wandb.ai/cal-capstone/SEN12MS/table?workspace=user-taeil)

**Comment by taeil** [comment]
https://wandb.ai/cal-capstone/SEN12MS
